src/iuse_actor.cpp

```diff
@@ -2583,6 +2599,16 @@ repair_item_actor::attempt_hint repair_item_actor::repair( player &pl, item &too
     if( action == RT_REPAIR ) {
         if( roll == SUCCESS ) {
             pl.add_msg_if_player(m_good, _("You repair your %s!"), fix.tname().c_str());
+            if( t->has_quality( quality_sew ) ) {
+                p->add_msg_if_player(m_info,
+                    _("You have used %d filament charges."), filament_required);
+                for (const auto &it : req.get_components()) {
+                    p->consume_items(it);
+                }
+                for (const auto &it : req.get_tools()) {
+                    p->consume_tools(it);
+                }
```

Comment:
> Game asks for filament only on SUCCESS roll, not on every tick - that is exactly the same like handle_components function is being called.

Well, that's wrong: the current master behavior is that thread is consumed every attempt.
What happens on the turns it isn't consumed?

Also, repair actor is used for soldering too. What happens here?

You should bring charges back to tailoring tools and only have them do the advanced sewing without them for now. Otherwise expect it to take much longer as you'll have to handle soldering too.